scipy/stats/_stats_py.py

```diff
@@ -4642,28 +4642,83 @@ def fisher_exact(table, alternative='two-sided'):
+           systematic review and meta-analysis."
+           BMJ, 345, :doi:`10.1136/bmj.e6779`, 2012.
+
     Examples
     --------
-    Say we spend a few days counting whales and sharks in the Atlantic and
-    Indian oceans. In the Atlantic ocean we find 8 whales and 1 shark, in the
-    Indian ocean 2 whales and 5 sharks. Then our contingency table is::
+    In [3]_, the effective dose of acetazolamide for the prophylaxis of acute
+    mountain sickness was investigated. The study notably concluded:
+
+        Acetazolamide 250 mg, 500 mg, and 750 mg daily were all efficacious for
+        preventing acute mountain sickness. Acetazolamide 250 mg was the lowest
+        effective dose with available evidence for this indication.
+
+    Identifying the correct effective dose is particularly important as
+    side effects to acetazolamide are frequent.
+
+    The following table summarizes the results of the experiment in which
+    some participants took a daily dose of acetazolamide 250 mg while others
+    took a placebo.
+    Cases of acute montain sickness were recorded::
+
+                                    Acetazolamide   Control/Placebo
+        Acute montain sickness             7           17
+        No                                15            5
+
+    Is there evidence that the acetazolamide 250 mg reduces the risk of
+    acute montain sickness?
+    We begin by formulating a null hypothesis :math:`H_0`:
 
-                Atlantic  Indian
-        whales     8        2
-        sharks     1        5
+        The effect of acetazolamide is equivalent to that of placebo.
 
-    We use this table to find the p-value:
+    Let's assess the plausibility of this hypothesis with
+    Fisher's test.
 
     >>> from scipy.stats import fisher_exact
-    >>> res = fisher_exact([[8, 2], [1, 5]])
+    >>> res = fisher_exact([[7, 17], [15, 5]])
+    >>> res.statistic
+    0.13725490196078433
     >>> res.pvalue
-    0.0349...
+    0.005768386750469949
+
+    Using a significance level of 5%, we would reject the null hypothesis in
+    favor of the alternative hypothesis: "the effect of acetazolamide
+    is not equivalent to the effect of placebo and reduces the risk of
+    acute montain sickness."
+
+    .. note::
+
+        Fisher's test assumes a double conditioning on the marginals.
+        It means that the sums of the columns and rows should be fixed
+        as parameters of the design of the experiment.
+
+        In this case, the sum of columns is fixed. 22 persons are in both
+        groups. But the number of person getting sick in both groups is not
+        (and cannot be) fixed before conducing the experiment. It is a
+        consequence.
+
+        As one marginal is not fixed, the row sum, Fisher's test will be less
+        powerful, more conservative. We can compare the result with
+        `boschloo_exact` which is a more powerful test and does not assume
+        any conditioning on the marginals.
```

Comment:
1. I don't understand the terminology with "conditional"/"unconditional". The null distribution of Boschloo's test is derived under the condition that the column sums (or rows, depending on the convention of the table) are fixed, so I would have expected it to be described as "conditioned" on the column marginals and "not conditioned" on the row marginals. yet it is described as "unconditional". Do you know where my understanding goes wrong? Unless we can get to the bottom of that, I'd suggest we use the commonly-understood phrase "unconditional exact test" rather than  "it does not assume any conditioning on the marginals".
2. Is it common to refer to the sums of both rows as "one marginal"? I am familiar with speaking in terms of two row marginals and two column marginals, or two sets of marginals - a set of row marginals, and a set of column marginals. (Of course, if the total is fixed, and one row marginal is fixed, then the other is fixed, too.) I would say "As the row marginals are fixed, ..."
3. The way this sentence is constructed, I think it implies "As [the row sums are not fixed], Fisher's test will be less powerful [and] more conservative [**than it would be if both row and column sums were fixed**]". I don't think that's the point you mean to make. I think you mean to make a statement about the power of Fisher's test compared to a test that is not conditioned on both marginals (e.g. Boschloo).

I think the following would be accurate and more in line with the point being made:
> Because the null distribution of Fisher's exact test is formed under the assumption that both row and column sums are fixed, the result of the test are conservative when applied to an experiment in which the row sums are not fixed. Boschloo's test does not depend on the assumption that the row sums are fixed, and consequently, it provides a more powerful test in this situation.

Reply:
Thanks, I made a mistake here. Boschloo needs a single conditioning (either row or col fixed).

Your version explains things in a easier way, let's go with that 👍

Reply:
I don't think it was necessarily a mistake, since Boschloo is described as "unconditional" in the literature (and Wikipedia). I have always assumed that "unconditional" was just in reference only to the row marginals; they don't bother to specify that it's conditioned on the column marginals because the case in which the column marginals are fixed is overwhelmingly common. (But another possible explanation is that my understanding of the word "conditional" might be wrong.)

But thanks; I think there's less chance that this version can be criticized as incorrect.